api/lib/domain/events/handle-complementary-certifications-scoring.js

```diff
@@ -30,60 +33,153 @@ async function handleComplementaryCertificationsScoring({
-          hasAcquiredPixCertification: assessmentResult.isValidated(),
-          minimumEarnedPix,
-          minimumReproducibilityRate,
-          isRejectedForFraud: certificationCourse.isRejectedForFraud(),
-        });
+          reproducibilityRate: assessmentResult.reproducibilityRate,
+          minimumEarnedPixForCurrentLevel: minimumEarnedPix,
+          minimumEarnedPixForLowerLevel: badgeNextLowerLevel.minimumEarnedPix,
+          complementaryCertificationScoring,
+          minimumReproducibilityRateLowerLevel,
+        })
+      ) {
+        return await _saveLowerLevel({ complementaryCertificationScoring, badgeNextLowerLevel });
+      }
+      return await _saveCurrentLevel(
+        complementaryCertificationCourseResultRepository,
+        complementaryCertificationScoring,
+      );
+    } else {
+      return await _saveCurrentLevel(
+        complementaryCertificationCourseResultRepository,
+        complementaryCertificationScoring,
+      );
     }
+  }
 
+  async function _saveLowerLevel({ complementaryCertificationScoring, badgeNextLowerLevel }) {
     await complementaryCertificationCourseResultRepository.save(
       ComplementaryCertificationCourseResult.from({
-        ...complementaryCertificationScoringWithComplementaryReferential,
+        ...complementaryCertificationScoring,
+        complementaryCertificationBadgeId: badgeNextLowerLevel.id,
         source: ComplementaryCertificationCourseResult.sources.PIX,
-        acquired: complementaryCertificationScoringWithComplementaryReferential.isAcquired(),
+        acquired: true,
       }),
     );
   }
 }
 
-function _buildComplementaryCertificationScoringWithReferential(
+async function _saveCurrentLevel(complementaryCertificationCourseResultRepository, complementaryCertificationScoring) {
+  await complementaryCertificationCourseResultRepository.save(
+    ComplementaryCertificationCourseResult.from({
+      ...complementaryCertificationScoring,
+      source: ComplementaryCertificationCourseResult.sources.PIX,
+      acquired: complementaryCertificationScoring.isAcquired(),
+    }),
+  );
+}
+
+function _hasAcquiredLowerLevel({
+  pixScore,
+  reproducibilityRate,
+  minimumEarnedPixForCurrentLevel,
+  minimumEarnedPixForLowerLevel,
+  complementaryCertificationScoring,
+  minimumReproducibilityRateLowerLevel,
+}) {
+  if (pixScore < minimumEarnedPixForLowerLevel) {
+    return false;
+  }
+  if (reproducibilityRate < minimumReproducibilityRateLowerLevel) {
+    return false;
+  }
+  return (
+    (_isBelowMinimumCurrentLevelScore({ pixScore, minimumEarnedPixForCurrentLevel }) &&
+      _isAboveMinimumLowerLevelScore({ pixScore, minimumEarnedPixForLowerLevel })) ||
+    (_isBelowMinimumReproducibilityRate(complementaryCertificationScoring) &&
+      _isAboveMinimumLowerReproducibilityRate(complementaryCertificationScoring, minimumReproducibilityRateLowerLevel))
+  );
+}
+
+function _isBelowMinimumReproducibilityRate(complementaryCertificationScoring) {
```

Comment:
Au risque d'être trop verbeux, j'aurais apporté une précision supplémentaire comme dans la fonction de dessous.

```suggestion
function _isBelowMinimumCurrentLevelRequiredReproducibilityRate(complementaryCertificationScoring) {

```